resources/assets/components/MultiValueFilter/index.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { map } from 'lodash';
+import classnames from 'classnames';
+
+import '../Tags/tags.scss';
+
+class MultiValueFilter extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      'good-photo': false,
```

Comment:
Something for thought, we keep defining these tag key/value pairs in different places. Is there a way to use the object that is passed into the `MultivalueFilter` component and pass them up to parents as needed.

Reply:
And actually this `multivalueFilter` really shouldn't care what is passed into it, so defining these properties on the state sorta confines it's usage. What if when one of the filter tags was clicked, it just pushed it's key/value pair to the state.

Reply:
The only reason why I defined the default state here so the button would know if it was active or not for the below (`{'is-active': this.state[key]}`) so the front end knows if it should have a background of blue if it is clicked and white if it is not clicked. Is it ok to keep this?

Reply:
hmm maybe the component should take in an extra property value if the key/value pair is active or not. So like `CampaignSingle` knows what it is using to grab posts and maybe it should tell the `MultivalueFilter` that along with what options to render. We want to move towards these components being as agnostic of the content they are rendering as possible. So this just can just be a component that has buttons and allows the user to select multiple of them.

Reply:
hmm I'm a little confused by this - so `CampaignSingle` would pass something like `options={{'good-photo': {label: 'Good Photo', active: false}, ...}}`?

Reply:
ya! that's what i'm thinking.

Reply:
ah got it - I'll try to fidget with that!

Reply:
@sbsmith86 sorry for all the questions! But just want to make sure I'm understanding correctly. We'll have to be able to change `active` to re-render showing if the tag is active or not so does that mean that this should be added to the `state` in the `CampaignSingle` page  and then passed down as `options={{'good-photo': {label: 'Good Photo', active: this.state.tags.good-photo.active}, ...}}`?

Reply:
yea probably. you could have a `filters` property on the state in `CampaignSingle` that the `FilterBar` updates when the user submits the filters to make the api request. It could also be used to determine what the active filters are.

Reply:
I think this would be an issue though because then, the tag buttons would only turn blue once you submit and update the state, not when you click on the actual tag. For some reason, I'm trying to update the state in the `MultiValueFilter` component but it doesn't seem to have access to the state from `CampaignSingle` :-/

Reply:
I updated the gist example to show what I am thinking! Check out the updates in `CampaignSingle` and If this really won't work, we can come back to it later. Basically you use the `CampaignSingle` state `tags` proper to render the `MultivalueFilter` component. This will allow React to trigger a re-render of the component when the state gets updated. https://gist.github.com/sbsmith86/f75bd68b6aada49cb35ae090a0c199dd

Reply:
Thanks so much for doing all of this, @sbsmith86 ! So I think the problem is that it still won't update the tag to be active until you hit the submit/filter button. Since the `FilterBar` has it's own state, we won't be able to update `active` in the state until `getPostsByFilter`, which is reached after the submit/filter button is pushed and when it re-renders to pass the state down to the `MultiValueFilter` button to tell if the tag is active or not. Does that make sense? I'm trying to find a way to pass down the state to `FilterBar` so we can update it there in the `updateFilters` function but not sure if you can do this?

Reply:
Maybe we can move `updateFilters` to the `CampaignSingle` page though?

Reply:
Actually I don't think that makes sense...

Reply:
That does make sense! but I think I don't understand why that is an issue? When `CampaignSingle` mounts, or grabs a new set of posts, it should update it's own state with the filters it used (including setting if they are active or not) whenever `CampaignSingle` updates it's state it will trigger a re-render down the tree, so `MultvalueFilter` will re-render with the new active/not active values.

Reply:
I think because it doesn't actually update the state of the tag in `updateFilters` since it doesn't have access to the state from `CampaignSingle`? 

Here's a giphy to show more what is happening (when tags are clicked, they aren't active but correct results are shown):
![sep-15-2017 15-50-56](https://user-images.githubusercontent.com/9019452/30500816-04d3fad0-9a2e-11e7-8860-7e9a63f1cedd.gif)

Reply:
I can also push up what I have so far if that is helpful!

Reply:
wait @sbsmith86 I think I got it working!! Thanks for your patience with me on this!

Reply:
Well I spoke too soon but getting there! I think I'll push up in a bit to walk through the problem I'm still having!

Reply:
ok!